### PR TITLE
Refactor: Allow specific passes to be dumped as fortran code

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

File: src/libasr/pass/pass_manager.h [hunk 1]
```diff
@@ -172,14 +172,13 @@ namespace LCompilers {
...
+                        outfile << "! Fortran code after applying the pass: " << passes[i]
+                            << "\n" << fortran_code.result << "\n";
+                        outfile.close();
                     }
-                    std::string str_i = std::to_string(i+1);
-                    if ( i < 9 )  str_i = "0" + str_i;
-                    std::ofstream outfile ("pass_" + str_i + "_" + passes[i] + ".f90");
-                    outfile << "! Fortran code after applying the pass: " << passes[i]
-                        << "\n" << fortran_code.result << "\n";
-                    outfile.close();
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/pass_manager.h]: I think `throw LCompilersException` itself halts the program execution, so having an else does not seem to be a compulsion. I would remove the `else {}` here and have the code in the same scope as that of `LCompilers::Result<std::string> fortran_code = asr_to_fortran(*asr, diagnostics, false, 4);` there by saving an extra indentation/horizontal space.

**Review by Shaikh-Ubaid — APPROVED** [review]
I reviewed this commit by commit. I shared a minor comment.

The changes look good to me! Thank you so much for this!

**Review by Pranavchiku — APPROVED** [review]
Awesome! Thanks for doing this, it is much required feature!

File: src/libasr/pass/pass_manager.h (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply src/libasr/pass/pass_manager.h]: Sure, I will remove the else block.

Comment (Thirumalai-Shaktivel) [reply src/libasr/pass/pass_manager.h]: Done!

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the reviews!

**Pull request merged**